reference/robot.md

```diff
@@ -416,268 +814,988 @@ The `wb_robot_get_battery_sampling_period` function returns the period given int
-This sample world is located in the "projects/samples/demos/worlds" directory of Webots.
+This function returns the value of the `basicTimeStep` field of the [WorldInfo](worldinfo.md) node.
 
 ---
 
-**Name**
+#### `wb_robot_get_mode`
 
-**wb\_robot\_get\_model** - *return the model defined in the robot node*
+%tab-component
 
-{[C++](cpp-api.md#cpp_robot)}, {[Java](java-api.md#java_robot)}, {[Python](python-api.md#python_robot)}, {[Matlab](matlab-api.md#matlab_robot)}, {[ROS](ros-api.md)}
+%tab "C"
 
 ```c
 #include <webots/robot.h>
 
-const char *wb_robot_get_model();
+<<<<<<< HEAD
+WbRobotMode wb_robot_get_mode();
+void wb_robot_set_mode(WbRobotMode mode, void *arg);
+=======
+typedef enum {
+  WB_MODE_SIMULATION,
+  WB_MODE_CROSS_COMPILATION,
+  WB_MODE_REMOTE_CONTROL
+} WbRobotMode;
+
+int wb_robot_get_mode();
+void wb_robot_set_mode(int mode, void *arg);
+>>>>>>> origin/webots-r2018a-revision-3
 ```
 
-**Description**
+%tab-end
 
-This function returns the model string as it is defined in the model field of the robot node (Robot, DifferentialWheels, Supervisor, etc.) in the current world file.
-The string returned should not be deallocated, as it was allocated by the "libController" shared library and will be deallocated when the controller terminates.
+%tab "C++"
 
----
+```cpp
+#include <webots/Robot.hpp>
 
-**Name**
+namespace webots {
+  class Robot {
+    enum {
+      MODE_SIMULATION,
+      MODE_CROSS_COMPILATION,
+      MODE_REMOTE_CONTROL
+    };
+
+    int getMode() const;
+    void setMode(int, void *);
+    // ...
+  }
+}
+```
 
-**wb\_robot\_get\_custom\_data**, **wb\_robot\_set\_custom\_data** - *return the custom data defined in the robot node*
+%tab-end
 
- - *set the data defined in the robot node*
+%tab "Python"
 
-{[C++](cpp-api.md#cpp_robot)}, {[Java](java-api.md#java_robot)}, {[Python](python-api.md#python_robot)}, {[Matlab](matlab-api.md#matlab_robot)}, {[ROS](ros-api.md)}
+```python
+from controller import Robot
 
-```c
-#include <webots/robot.h>
+class Robot:
+    MODE_SIMULATION, MODE_CROSS_COMPILATION, MODE_REMOTE_CONTROL
 
-const char * wb_robot_get_custom_data();
-void wb_robot_set_custom_data(const char *data);
+    def getMode(self):
+    # ...
 ```
 
-**Description**
+%tab-end
 
-The `wb_robot_get_custom_data` function returns the string contained in the `customData` field of the robot node.
+%tab "Java"
 
-The `wb_robot_set_custom_data` function set the string contained in the `customData` field of the robot node.
+```java
+import com.cyberbotics.webots.controller.Robot;
 
----
+public class Robot {
+  public final static int MODE_SIMULATION, MODE_CROSS_COMPILATION, MODE_REMOTE_CONTROL;
 
-**Name**
+  public int getMode();
```

Comment:
setMode is missing.

Reply:
setMode() functions are problematic in Python and Java because of the `void *arg` parameter. I think they are not documented on purpose.